packages/iTwinUI-react/src/core/ComboBox/ComboBox.tsx

```diff
@@ -169,16 +163,41 @@ export const ComboBox = <T,>(props: ComboBoxProps<T>) => {
     optionsExtraInfoRef.current = {};
```

Comment:
Maybe you can move this to your newly created useEffect?

Reply:
moved